Atapiave/QA_team
# FEAT: Add new login function

## Any background context you want to provide?

A login method is required so all apps can start creating e2e flows.

## What does this PR do?

Adds a function that will allow all QAs to login to the app so they can start creating e2e flows


## Where should the reviewer start?

In the following file:

* Behave/screens/login/login_screen.py
Here is where the login function is present.

* Behave/utils/dictionaries/os_locators.py
Here is where the locators for interacting with the OS prompts are going to be placed. For now this only has the selectors for the Permissions prompts.

* Behave/screens/base_actions.py *
Here we added 2 new methods, 1 for allow the app to access the device's location and another one to deny it.


## How should this be tested?

By creating an object of the LoginScreen class and using the login method. Or by running the existing Login test as follows:

1.- Clone this branch
2.- Create an .env file by creating a copy of the .env-sample
3.- Fill the .env file with the details of a user that exists in the DEV environment and other required details like:
`OTP_API_URL = 'http://public-cluster-fargate-superapp-259682327.us-east-1.elb.amazonaws.com/superapp/qa/otp/v1/'`
4.- Add an android _baz-dev.apk_ inside the Behave/APP/ folder
5.- Open a Terminal window (Could be inside of the IDE you are using)
6.- Get inside the Behave folder
7.- Run the following command:
`paver run_feature android <YourDeviceID> login   `
8.- Check in your physical phone that the login process is being executed.

## What are the relevant tickets?

TBD

## Questions :question:
#### Checklist
